ClickHouse/ClickHouse
# fix part mutation

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (up to few sentences, required except for Non-significant/Documentation categories):
Fixed the bug that mutations be skipped for some attached parts due to their data_version are larger than the table mutation version.

Detailed description (optional):
## How to reproduce
- Create a table with two parts, do a mutation. 
- Backup it with clickhouse-backup or do `FREEZE` manually. 
- Drop the table. 
- Restore the table with clickhouse-backup or do `CREATE` and `ATTACH` manually. 
- Do another mutation.
- Verify if the mutation take effect. 

...
CREATE TABLE test_orders(
    order_id String,
    order_time DateTime,
    amount Float32
) ENGINE = MergeTree()
PARTITION BY toYYYYMMDD(order_time)
ORDER BY (order_time, order_id);

INSERT INTO test_orders(order_id, order_time, amount) VALUES ('1', '0000-00-00 00:00:00', 10),  ('2', '2019-01-01 00:00:00', 10);

ALTER TABLE test_orders UPDATE amount=amount*2 WHERE order_time>='2019-01-01 00:00:00';

-- sleep 3;
SELECT count(), countIf(amount>15) FROM test_orders;
┌─count()─┬─countIf(greater(amount, 15))─┐
│       2        │                                               1 │
└──────┴─────────────────────┘

sudo clickhouse-backup create -t default.test_orders test

DROP TABLE test_orders;

sudo clickhouse-backup restore -t default.test_orders test

SELECT count(), countIf(amount>15) FROM test_orders;
┌─count()─┬─countIf(greater(amount, 15))─┐
│       2 │                            1 │
└─────────┴──────────────────────────────┘

ALTER TABLE test_orders UPDATE amount=10 WHERE 1;

SELECT count(), countIf(amount>15) FROM test_orders;
┌─count()─┬─countIf(greater(amount, 15))─┐
│       2 │                            1 │
└─────────┴──────────────────────────────┘

However I expect:
┌─count()─┬─countIf(greater(amount, 15))─┐
│       2 │                            0 │
└─────────┴──────────────────────────────┘


...
